database/conversations.js

```diff
@@ -1,49 +1,93 @@
-import { db } from './database.js';
+const { dbRun, dbGet, dbAll } = require('./database');
 
 // Store a recorded conversation
-function storeConversation(chatbotId, conversation, pageUrl, date) {
-    return new Promise((resolve, reject) => {
-      db.run(
-        `INSERT INTO recorded_conversation (chatbot_id, conversation, page_url, date) VALUES (?, ?, ?, ?)`,
-        [chatbotId, conversation, pageUrl, date],
-        function (err) {
-          if (err) reject(err);
-          else resolve(this.lastID);
-        }
-      );
-    });
-  }
-  
-  // Retrieve conversations for a chatbot
-  function getConversationsByChatbot(chatbotId) {
-    return new Promise((resolve, reject) => {
-      db.all(
-        `SELECT * FROM recorded_conversation WHERE chatbot_id = ?`,
-        [chatbotId],
-        (err, rows) => {
-          if (err) reject(err);
-          else resolve(rows);
-        }
-      );
-    });
-  }
-  
-  // Delete a recorded conversation
-  function deleteConversation(recordedConversationId) {
-    return new Promise((resolve, reject) => {
-      db.run(
-        `DELETE FROM recorded_conversation WHERE recorded_conversation_id = ?`,
-        [recordedConversationId],
-        function (err) {
-          if (err) reject(err);
-          else resolve();
-        }
-      );
-    });
+async function storeConversation(chatbotId, conversation, pageUrl, date, chatId) {
+    // Check if there's an existing conversation with this chat_id
+    const existingConversation = await dbGet(
+        `SELECT recorded_conversation_id, conversation FROM recorded_conversations 
+         WHERE chat_id = ?`,
+        [chatId]
+    );
+
+    if (existingConversation) {
+        // Update existing conversation
+        return await dbRun(
+            `UPDATE recorded_conversations 
+             SET conversation = ?, date = ?
+             WHERE recorded_conversation_id = ?`,
+            [JSON.stringify(conversation), date, existingConversation.recorded_conversation_id]
+        );
+    } else {
+        // Create new conversation
+        return await dbRun(
+            `INSERT INTO recorded_conversations (chatbot_id, conversation, page_url, date, chat_id) 
+             VALUES (?, ?, ?, ?, ?)`,
+            [chatbotId, JSON.stringify(conversation), pageUrl, date, chatId]
+        );
+    }
+}
+
+// Retrieve conversations for a chatbot with pagination and filters
+async function getConversationsByChatbot(chatbotId, page = 1, limit = 10, dateFilter = null, pageFilter = '') {
```

Comment:
We should confirm that the user actually owns this chatbot by using the auth middleware function defined in website-api/middleware.js